### ISSUE TITLE
Symlink treated as directory

### ISSUE THREAD

**Issue by andreasabel** [ndmitchell/filepattern#5]
This is 
- https://github.com/agda/fix-whitespace/issues/9

Reproducer in `filepattern-0.1.2`:
```
$ mkdir dir
$ mkdir dir/a
$ mkdir dir/b
$ touch dir/a/file
$ ln -s dir/a/file dir/b/file
$ tree
.
└── dir
    ├── a
    │   └── file
    └── b
        └── file -> dir/a/file
$ ghci
GHCi, version 9.2.3: https://www.haskell.org/ghc/  :? for help
ghci> :m + System.FilePattern.Directory
ghci> getDirectoryFiles "." ["**"]
*** Exception: ./dir/b/file/: getDirectoryContents:openDirStream: does not exist (No such file or directory)
```

**Comment by andreasabel** [comment]
Relevant code: https://github.com/ndmitchell/filepattern/blob/1fe716c21feabcc3d168390814dd6f107ea5f14d/src/System/FilePattern/Directory.hs#L94-L101
In https://github.com/ndmitchell/filepattern/blob/1fe716c21feabcc3d168390814dd6f107ea5f14d/src/System/FilePattern/Directory.hs#L100
we get `isFile == Just False` for a symlink, which is then treated as directory.
The docs for `doesFileExist` _are_ misleading (outright wrong):
https://hackage.haskell.org/package/directory-1.3.7.1/docs/System-Directory.html#v:doesFileExist
> The operation [doesFileExist](https://hackage.haskell.org/package/directory-1.3.7.1/docs/System-Directory.html#v:doesFileExist) returns [True](https://hackage.haskell.org/package/base-4.14.1.0/docs/Data-Bool.html#v:True) if the argument file exists and is not a directory, and [False](https://hackage.haskell.org/package/base-4.14.1.0/docs/Data-Bool.html#v:False) otherwise.

**Comment by andreasabel** [comment]
Since `if doesExist && isFile == Just False then pure True` isn't working for symlinks, we need to check `doesDirectoryExist` anyway, thus can get rid of the `doesExist` optimization altogether. Implemented in:
- #6

**Comment by andreasabel** [comment]
Ping @ndmitchell.

**Comment by ndmitchell** [comment]
Thanks for the diff @andreasabel - unfortunately you've caught me as I'm travelling and otherwise unavailable. Hopefully will take a proper look at this tomorrow.

**Comment by andreasabel** [comment]
Thanks for merging!  Please ping me when you release, so I can depend on the latest version.

**Comment by ndmitchell** [comment]
@andreasabel 0.1.3 now released with this fix

**Comment by andreasabel** [comment]
Thanks, published https://hackage.haskell.org/package/fix-whitespace-0.0.10.
